inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -0,0 +1,1348 @@
+        blk_size = 16;
+    } else if (mayiuse(cpu::avx2)) {
+        reduce_kernel.reset(new jit_uni_reduce_kernel_f32<cpu::avx2>(jcp));
+        blk_size = 8;
+    } else if (mayiuse(cpu::sse42)) {
+        reduce_kernel.reset(new jit_uni_reduce_kernel_f32<cpu::sse42>(jcp));
+        blk_size = 8;
+    }
+}
+
+void MKLDNNReduceNode::execute(mkldnn::stream strm) {
+    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto &srcMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
+    auto &srcIndexesMemPtr = getParentEdgeAt(REDUCE_INDEXES)->getMemoryPtr();
+
+    const auto idx_data = reinterpret_cast<const int32_t *>(srcIndexesMemPtr->GetData());
+    size_t dst_size = dstMemPtr->GetSize();
+    src_dims = getParentEdgeAt(REDUCE_DATA)->getDesc().getDims();
+    src_strides = getParentEdgeAt(REDUCE_DATA)->getDesc().getBlockingDesc().getStrides();
+    dims_size = src_dims.size();
+    calc_process_dst_dims(idx_data);
+
+    if (dims_size <= 5) {
+        if (dims_size == 5) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], src_dims[2], src_dims[3], src_dims[4]);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], process_dst_dims[2], process_dst_dims[3], process_dst_dims[4]);
+        } else if (dims_size == 4) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], 1, src_dims[2], src_dims[3]);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], 1, process_dst_dims[2], process_dst_dims[3]);
+        } else if (dims_size == 3) {
+            SET_SRC_DIM_VALUE(1, src_dims[0], 1, src_dims[1], src_dims[2]);
+            SET_DST_DIM_VALUE(1, process_dst_dims[0], 1, process_dst_dims[1], process_dst_dims[2]);
+        } else if (dims_size == 2) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], 1, 1, 1);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], 1, 1, 1);
+        } else {
+            SET_SRC_DIM_VALUE(1, src_dims[0], 1, 1, 1);
+            SET_DST_DIM_VALUE(1, process_dst_dims[0], 1, 1, 1);
+        }
+
+        ReduceN = IB != OB && OB == 1;
+        ReduceC = IC != OC && OC == 1;
+        ReduceD = ID != OD && OD == 1;
+        ReduceH = IH != OH && OH == 1;
+        ReduceW = IW != OW && OW == 1;
+    }
+
+    auto src_data_offset = srcMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding *
+         MKLDNNExtensionUtils::sizeOfDataType(mkldnn::memory::data_type(srcMemPtr->GetDescriptor().data.data_type));
+    auto dst_data_offset = dstMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding *
+         MKLDNNExtensionUtils::sizeOfDataType(mkldnn::memory::data_type(dstMemPtr->GetDescriptor().data.data_type));
+    if (input_prec == Precision::FP32) {
+        auto src_data = reinterpret_cast<const float *>(srcMemPtr->GetData()) + src_data_offset;
+        if (output_prec == Precision::FP32) {
+            auto dst_data = reinterpret_cast<float *>(dstMemPtr->GetData()) + dst_data_offset;
```

Comment:
src_data_offset and dst_data_offset is offset size by byte. In line 674 and 676, you cast ptr to float and then add offset, leading to advance 4*offset actually. Same for following lines.

Reply:
you can refer: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L928

Reply:
Fixed. Thanks Chenhu!